code/game/supplyshuttle.dm

```diff
@@ -46,6 +46,34 @@ var/list/mechtoys = list(
 	layer = 4
 	explosion_resistance = 5
 
+/obj/structure/plasticflaps/New()
+	..()
+	anchored=0
+
+/obj/structure/plasticflaps/attackby(obj/item/I as obj, mob/user as mob)
+	..()
+	if (istype(I, /obj/item/weapon/crowbar))
+		if(anchored == 1)
+			user.visible_message("[user] pops loose the flaps.", "You pop loose the flaps.")
+			anchored = 0
+			var/turf/T = get_turf(loc)
+			if(T)
+				T.blocks_air = 0
+		else
+			user.visible_message("[user] pops in the flaps.", "You pop in the flaps.")
+			anchored = 1
+			var/turf/T = get_turf(loc)
+			if(T)
+				T.blocks_air = 1
+	else if (iswelder(I) && anchored == 0)
+		var/obj/item/weapon/weldingtool/WT = I
+		if(WT.remove_fuel(0, user))
+			var/obj/item/stack/sheet/mineral/plastic/P = new /obj/item/stack/sheet/mineral/plastic (user.loc)

```

Comment:
I'd suggest using src.loc, since it makes more sense.
Amount is also the second argument of New for sheets, so it makes more sense to just write new/obj/item/stack/sheet/mineral/plastic(src.loc, amount = 10) and not use a var.